### PR TITLE
POLIO 661: campaign history UI improvements

### PR DESCRIPTION
**Pull request by codeKameleon**
Poor readibility of the the data coming from logs especially for nested values (rounds, scopes,...)
This PR was made in pair coding with @beygorghor to improve the UI

Related JIRA tickets : [POLIO-661](https://bluesquare.atlassian.net/browse/POLIO-661?atlOrigin=eyJpIjoiOTYwMzJlY2M2YmZlNDUxNjkwMzYyZDIyMGM4ZjQwOTciLCJwIjoiaiJ9)

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [x] New translations have been added or updated if new string have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- add `ExpandableItem` component for complex values with nested objects and arrays (rounds, scopes, vaccines, shipments, destructions)
- create a `config.tsx `file to get the key and convert it if necessary with a `getLogValue` function
- centralize logic to render rows with array/objects in `mapLogStructure` function

## How to test

- Go to Campaign
- Choose a campaign
- Edit some fields
- Click on history icon in the edit dialog
- Check through all the log if values are well displayed (ExpandableItem for nested values, links should work, date in the right format)

## Print screen / video

https://user-images.githubusercontent.com/25134301/226388947-b91e23e0-c662-4bbd-8402-c3c8c6ed2259.mov

[POLIO-661]: https://bluesquare.atlassian.net/browse/POLIO-661?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: plugins/polio/js/src/components/campaignHistory/CampaignLogDetail.tsx [hunk 1]
```diff
@@ -42,19 +26,6 @@ export type Result = {
...
-    },
-    linkToChangesLog: {
-        color: theme.palette.primary.main,
-        textAlign: 'right',
-        flex: '1',
-        cursor: 'pointer',
-    },
-}));
-
 const objectLoop = (obj, getValue, getLabel) => {
```
Comment (beygorghor) [inline plugins/polio/js/src/components/campaignHistory/CampaignLogDetail.tsx]: This part of the code is not used anymore i guess

Comment (beygorghor) [reply plugins/polio/js/src/components/campaignHistory/CampaignLogDetail.tsx]: objectLoop, ArrayLoop getComplexValue ...

**Review by beygorghor — CHANGES_REQUESTED** [review]
Coool , thanks for this ! but:
- you have dead code to remove
- This one is a bit alone, we can add the OBR name at least
Screenshot 2023-03-22 at 10 28 05
- I still have translatio
Screenshot 2023-03-22 at 10 28 30
ns warning in the console
-

File: plugins/polio/js/src/components/campaignHistory/CampaignHistory.tsx [hunk 2]
```diff
@@ -78,7 +78,7 @@ export const CampaignHistory: FunctionComponent<Props> = ({
         isError: boolean;
     } = useGetCampaignLogDetail(params.logId);
 
-    const { user: campaignUser } = useMemo(() => {
+    const { user: campaignUser, logDetail: campaignLogDetail } = useMemo(() => {
         if (!data) {
             return { user: undefined, logDetail: undefined };
         }
```
Comment (quang-le) [inline plugins/polio/js/src/components/campaignHistory/CampaignHistory.tsx]: This should go in the `select` of `useGetCampaignLogDetail`. It's part of the same logic. Also you wouldn't have to repeat your code in `CampaignLogDetail`, lines 39-45

File: plugins/polio/js/src/components/campaignHistory/Head.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    TableRow,
+    TableCell,
+    makeStyles,
+    Theme,
+} from '@material-ui/core';
+
+import MESSAGES from '../../constants/messages';
+
+const useStyles = makeStyles((theme: Theme) => ({
+    ...commonStyles(theme),
```
Comment (quang-le) [inline plugins/polio/js/src/components/campaignHistory/Head.tsx]: why the import ? You don't seem to use any className from `commonStyles`

File: plugins/polio/js/src/hooks/useGetCampaignFieldLabel.ts [hunk 4]
```diff
@@ -14,3 +15,14 @@ export const useGetCampaignFieldLabel = (): ((fieldKey: string) => string) => {
...
+export const useGetChildrenLabel = (): ((
+    childrenLabel: IntlMessage,
+) => string) => {
+    const { formatMessage } = useSafeIntl();
+    const getLabel = (childrenLabel): string => {
+        return formatMessage(childrenLabel);
+    };
+
+    return getLabel;
+};
```
Comment (quang-le) [inline plugins/polio/js/src/hooks/useGetCampaignFieldLabel.ts]: The function you return is equivalent to `formatMessage`, without adding any functionality as `useGetCampaignFieldLabel` above does, so you don't need it

File: plugins/polio/js/src/components/campaignHistory/Row.tsx [hunk 5]
```diff
@@ -45,18 +55,27 @@ const truncate = (string, limit) => {
...
+    cellWithMargin = true,
+}) => {
     const classes: Record<string, string> = useStyles();
     const getLabel = useGetCampaignFieldLabel();
     const regex = '^(http|https):\\/\\/';
     return (
         <TableRow className={classes.round}>
             {fieldKey && (
-                <TableCell className={classes.tableCellLabel}>
+                <TableCell className={classes.tableCellLabel} width={300}>
```
Comment (quang-le) [inline plugins/polio/js/src/components/campaignHistory/Row.tsx:69]: Line 67: classes.round does not exist

**Review by quang-le — CHANGES_REQUESTED** [review]
Looks very nice, quite a neat solution overall. Still some small improvements that can be done to code (see comments).

Still some issues in the interface:

- Boolean fields should display more useful information, like yes or no. 

Screenshot 2023-03-24 at 16 50 29

- I think we should show the name of the org unit instead of the id, or show both

Screenshot 2023-03-24 at 16 46 12

- There is a field translated as "Approuvé", I think there's something wrong with the json key:
Screenshot 2023-03-24 at 16 48 20

- There are some translations missing: 

Screenshot 2023-03-24 at 16 43 27
Screenshot 2023-03-24 at 16 47 09
Screenshot 2023-03-24 at 16 44 59
Screenshot 2023-03-24 at 16 44 14
Screenshot 2023-03-24 at 16 47 54
Screenshot 2023-03-24 at 16 47 41

- "Fioles" should be replaced with "Flacons". There was a PR for this a while ago, but we may have missed some

Screenshot 2023-03-24 at 16 43 50

- "Scopes" should be translated here:
Screenshot 2023-03-24 at 16 45 45

- We have the same translation for 2 keys:
Screenshot 2023-03-24 at 16 46 56
This one is tricky, because we switched one key for the other at some point in the project

Lots of nitpicking but it's mostly minor stuff. Nice job!

**Comment by beygorghor** [comment]
Still missing 3 translations in french please:

Screenshot 2023-03-27 at 09 06 31

File: plugins/polio/js/src/components/campaignHistory/CampaignLogDetail.tsx [hunk 6]
```diff
@@ -27,22 +31,14 @@ export type Result = {
 
 export const CampaignLogDetail: FunctionComponent<Props> = ({ logId }) => {
     const {
-        data,
+        data: { logDetail: campaignLogDetail } = initialLogDetail,
```
Comment (quang-le) [inline plugins/polio/js/src/components/campaignHistory/CampaignLogDetail.tsx:26]: I think if we do this we don't need the `initialData` in the `useSnackQuery`

File: plugins/polio/js/src/components/campaignHistory/CampaignHistory.tsx [hunk 7]
```diff
@@ -67,24 +69,18 @@ export const CampaignHistory: FunctionComponent<Props> = ({
         isFetching: boolean;
         isError: boolean;
     } = useGetCampaignLogs(params.campaignId);
-
     const {
-        data,
+        data: {
+            user: campaignUser,
+            logDetail: campaignLogDetail,
+        } = initialLogDetail,
```
Comment (quang-le) [inline plugins/polio/js/src/components/campaignHistory/CampaignHistory.tsx:76]: see comment below

File: plugins/polio/js/src/hooks/useGetCampaignHistory.ts [hunk 8]
```diff
@@ -46,26 +42,37 @@ export const useGetCampaignLogs = (
...
+    initialData: CampaignLogDetailResult,
     logId?: string,
-): UseQueryResult<CampaignLogDetailResult | undefined, Error> => {
+): UseQueryResult<CampaignLogDetailResult, Error> => {
     return useSnackQuery({
         queryKey: ['campaignLogDetail', logId],
         queryFn: () => getCampaignLogDetail(logId),
         options: {
             enabled: Boolean(logId),
+            initialData,
```
Comment (quang-le) [inline plugins/polio/js/src/hooks/useGetCampaignHistory.ts:64]: see comment obove

**Review by quang-le — APPROVED** [review]
Nothing blocking.

**Review by beygorghor — APPROVED** [review]

**Pull request merged**